src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+
+namespace System.Device.Pwm
+{
+    /// <summary>
+    /// Represents a PWM channel running on Unix.
+    /// </summary>
+    internal class UnixPwmChannel : PwmChannel
+    {
+        private readonly int _chip;
+        private readonly int _channel;
+        private double _frequency;
+        private double _dutyCyclePercentage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixPwmChannel"/> class.
+        /// </summary>
+        /// <param name="chip">The PWM chip number.</param>
+        /// <param name="channel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity of the PWM signal.</param>
+        public UnixPwmChannel(
+            int chip,
+            int channel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _chip = chip;
+            _channel = channel;
+            Validate();
+            Open();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
```

Comment:
How was this not a problem with the previous implementation?

Reply:
I'm not sure.  I went back and tried with original and appears to work.  The cold is very similar, but I could be overlooking something.  I can reproduce in PwmChannel and get the following if it doesn't wait long enough.

> Error: Access to the path '/sys/class/pwm/pwmchip0/pwm0/period' is denied.

Reply:
Note: I came across this and appears to be similar issue.

https://github.com/processing/processing/blob/37108add372272d7b1fc23d2500dce911c4d1098/java/libraries/io/src/processing/io/PWM.java#L81

Reply:
mm we should look into this a bit more, it makes me a bit uncomfortable to have a Thread.Sleep without really knowing if it is required. I suppose you are running as sudo given that this seems to be a permission issue? Maybe the file gets created right away but permissions aren't set so if not running as sudo then it doesn't work at first?

Reply:
@shaggygi have you tried the sync solution I've written in one of the other comments?

Reply:
@krwq no. i briefly read over, but was working on the other stuff.  i'll try to review tonight.